weblab_cg/_generator_chaste_cg.py

```diff
@@ -179,6 +192,30 @@ def _get_var_display_name(self, var):
             display_name = var.cmeta_id
         return display_name.replace('$', '__').replace('var_chaste_interface_', '', 1).replace('var_', '', 1)
 
+    def _partial_eval(self, equations, required_lhs):
+        # only do partial eval if pe option is set
+        if not self._pe:
+            return equations
+
+        evaluated_eqs = list(reversed(equations))
+        # count usage of variables on rhs of equations
+        usage_count = dict()
+        for i in range(len(evaluated_eqs) - 1, - 1, - 1):
+            usage_count[evaluated_eqs[i].lhs] = 0
+            for var in evaluated_eqs[i].rhs.atoms(VariableDummy):
+                usage_count[var] = usage_count[var] + 1 if var in usage_count else 1
```

Comment:
You may find the `dict.setdefault` method handy here.

Reply:
didn't know you could do that!